drydock_provisioner/drivers/oob/pyghmi_driver/__init__.py

```diff
@@ -300,4 +319,66 @@ def execute_task(self):
                 result=hd_fields.ActionResult.Success,
                 status=hd_fields.TaskStatus.Complete,
                 result_detail=mci_id)
-            return
\ No newline at end of file
+            return
+
+    def get_ipmi_session(self):
+        """
+        Initialize a Pyghmi IPMI session to this runner's self.node
+
+        :return: An instance of pyghmi.ipmi.command.Command initialized to nodes' IPMI interface
+        """
+
+        node = self.node
+
+        if node.oob_type != 'ipmi':
+            raise errors.DriverError("Node OOB type is not IPMI")
+
+        ipmi_network = self.node.oob_parameters['network']
+        ipmi_address = self.node.get_network_address(ipmi_network)
+
+        if ipmi_address is None:
+            raise errors.DriverError("Node %s has no IPMI address" %
+                (node.name))
+
+        
+        ipmi_account = self.node.oob_parameters['account']
+        ipmi_credential = self.node.oob_parameters['credential']
+
+        self.logger.debug("Starting IPMI session to %s with %s/%s" %
+                            (ipmi_address, ipmi_account, ipmi_credential[:1]))
+        ipmi_session = Command(bmc=ipmi_address, userid=ipmi_account,
+                               password=ipmi_credential)
+
+        return ipmi_session
+
+    def exec_ipmi_command(self, callable, *args):
+        """
+        Call an IPMI command after establishing a session with this runner's node
+
+        :param callable: The pyghmi Command method to call
+        :param args: The args to pass the callable
+        """
+        attempts = 0
+        while attempts < 5:
```

Comment:
Maybe it's better to break this up into 2 while loops? One for initialization and one for calling the command?